File: html/phpfunctions/insert_data.php
<?php

ini_set('display_errors', 1);
ini_set('display_startup_errors', 1);
error_reporting(E_ALL);

include_once('../system/init.php');
$DB = new DBPDO();

isset($_GET["device"]) ? $device = $_GET["device"] : $device = NULL;
isset($_GET["DI1"]) ? $DI1 = $_GET["DI1"] : $DI1 = NULL;
isset($_GET["DI2"]) ? $DI2 = $_GET["DI2"] : $DI2 = NULL;
isset($_GET["DI3"]) ? $DI3 = $_GET["DI3"] : $DI3 = NULL;
isset($_GET["DI4"]) ? $DI4 = $_GET["DI4"] : $DI4 = NULL;

isset($_GET["DO1"]) ? $DO1 = $_GET["DO1"] : $DO1 = NULL;
isset($_GET["DO2"]) ? $DO2 = $_GET["DO2"] : $DO2 = NULL;
isset($_GET["DO3"]) ? $DO3 = $_GET["DO3"] : $DO3 = NULL;
isset($_GET["DO4"]) ? $DO4 = $_GET["DO4"] : $DO4 = NULL;

isset($_GET["AI1"]) ? $AI1 = $_GET["AI1"] : $AI1 = NULL;
isset($_GET["AI2"]) ? $AI2 = $_GET["AI2"] : $AI2 = NULL;
isset($_GET["AI3"]) ? $AI3 = $_GET["AI3"] : $AI3 = NULL;
isset($_GET["AI4"]) ? $AI4 = $_GET["AI4"] : $AI4 = NULL;
isset($_GET["AI5"]) ? $AI5 = $_GET["AI5"] : $AI5 = NULL;

isset($_GET["AO1"]) ? $AO1 = $_GET["AO1"] : $AO1 = NULL;
isset($_GET["AO2"]) ? $AO2 = $_GET["AO2"] : $AO2 = NULL;
isset($_GET["AO3"]) ? $AO3 = $_GET["AO3"] : $AO3 = NULL;
isset($_GET["AO4"]) ? $AO4 = $_GET["AO4"] : $AO4 = NULL;
isset($_GET["AO5"]) ? $AO5 = $_GET["AO5"] : $AO5 = NULL;

$values = array($DI1, $DI2, $DI3, $DI4, $DO1, $DO2, $DO3, $DO4, $AI1, $AI2, $AI3, $AI4, $AI5, $AO1, $AO2, $AO3, $AO4, $AO5);


$DB->execute("INSERT INTO ".$device."  (DI1 , DI2, DI3, DI4, DO1, DO2, DO3, DO4, AI1, AI2, AI3, AI4, AI5, AO1, AO2, AO3, AO4, AO5) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) ", $values);

echo "hola";

File: jscab/insert_data.py
import socket
import time
import digitals
import analogs


#host = "192.168.1.42"  #ip address where is the web server running
host = "192.168.255.110"
port = 80
device = "raspberry2"  #name of this device



digitalData = []
lastDigitalData = []

analogInputs = []
lastAnalogInputs = [0,0,0,0,0]

analogOutputs = []
lastAnalogOutputs = [0,0,0,0,0]


		
def insertData(host, port, device, di1, di2, di3, di4, do1, do2, do3, do4, ai1, ai2, ai3, ai4, ai5, ao1, ao2, ao3, ao4, ao5):
	
		
	#create a socket
	comms_socket = socket.socket()
	print("socket created")
	
	try:
		#connect to host through that socket
		comms_socket.connect((host, port))
		print("connected to host")	
		
		#prepare the http request
		data = "GET /phpfunctions/insert_data.php?device=" + device + "&DI1=" + str(di1) + "&DI2=" + str(di2) + "&DI3=" + str(di3) + "&DI4=" + str(di4) + "&DO1=" + str(do1) + "&DO2=" + str(do2) + "&DO3=" + str(do3) + "&DO4=" + str(do4) + "&AI1=" + str(ai1) + "&AI2=" + str(ai2) + "&AI3=" + str(ai3) + "&AI4=" + str(ai4) + "&AI5=" + str(ai5) + "&AO1=" + str(ao1) + "&AO2=" + str(ao2) + "&AO3=" + str(ao3) + "&AO4=" + str(ao4) + "&AO5=" + str(ao5) + " HTTP/1.1\r\nhost: localhost\r\nConnection: close\r\n\r\n"
		
		#send the data	
		comms_socket.send(bytes(data, "UTF-8"))
		print("data sent")
					
		data_recv = comms_socket.recv(4096).decode("UTF-8")
		
		
		comms_socket.close()
		
		print(data_recv)
	except(BrokenPipeError, OSError):
		print ("conexion no establecida")


	
while True:	
	digitalData = digitals.readDigitals()
	analogInputs = analogs.readAnalogInputs(lastAnalogInputs)
	analogOutputs = analogs.readAnalogOutputs(lastAnalogOutputs)
		
	
	if (lastDigitalData != digitalData or lastAnalogInputs != analogInputs or lastAnalogOutputs != analogOutputs):
		insertData(host, port, device, digitalData[0], digitalData[1], digitalData[2], digitalData[3], digitalData[4], digitalData[5], digitalData[6],digitalData[7], analogInputs[0], analogInputs[1], analogInputs[2], analogInputs[3], analogInputs[4], analogOutputs[0], analogOutputs[1], analogOutputs[2], analogOutputs[3], analogOutputs[4])
		lastDigitalData = digitalData
		lastAnalogInputs = analogInputs
		lastAnalogOutputs = analogOutputs
		
